### ISSUE TITLE
After certain time leads to Redux warning "setState() on an unmounted component"

### ISSUE THREAD

**Issue by Restuta** [reactjs/react-redux#457]
I am testing v5 alpha from #416 and noticed that after some time every re-render in one of my components starts triggering following warning: 

`Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code `

My best guess is that there is a memory leak somewhere and new connect holds a reference to unmounted component. Triple checked that it doesn't happen with `4.4.5`. 

It doesn't happen right away, only after I switch back and forth between routes that re-render a lot of components. I don't have time to dig deeper and set up a test project, but I can create a branch in my public repo where it's happening to reproduce. Please let me know if you want this.

**Comment by max-mykhailenko** [comment]
I have the same issue

**Comment by jimbolla** [comment]
A repro would be very helpful.  Thanks.

**Comment by Restuta** [comment]
@max-mykhailenko do you have simple steps to repro?

**Comment by timdorr** [comment]
@Restuta Do you have a reproducible test case as well? 

Does this occur in the canonical `real-world` example? https://github.com/reactjs/redux/tree/master/examples/real-world

**Comment by Restuta** [comment]
@timdorr not it does not, I created a branch that you can use to repro it.

Branch: https://github.com/Restuta/rcn.io/tree/react-redux-connect-bug

Steps to repro:

* `git clone git@github.com:Restuta/rcn.io.git`
* cd `rcn.io`
* `npm i && npm run dev`
* open http://localhost:8888/events/8 in your browser
* open console
* click on MTB in the header and then back to "Event 8"

You should see in console:

"Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the Connect(EventDetails) component."

Notice that the only change from "dev" branch is bump of `redux-react` and you can test it yourself that issue is not happening with `4.4.5`.

**Comment by jimbolla** [comment]
Thanks,  I'll have a look at this tonight.

**Comment by jimbolla** [comment]
@Restuta That git repo address doesn't seem to work for me. Is it supposed to be `https://github.com/Restuta/rcn.io.git`?

**Comment by Restuta** [comment]
@jimbolla that address in SSH format, of you prefer HTTPS then yes, use that one

**Comment by jimbolla** [comment]
I have a fix ready for this [here](https://github.com/jimbolla/react-redux/tree/fix-457) but I'm going to attempt to write a test for it before submitting the PR. Honestly, writing the test is the hard part for me. :laughing:

**Comment by vhmth** [comment]
@jimbolla what do you think the timeline is for getting this in? This is the last thing before #465 gets in, which is causing some UI issues for our beta users so I'm more than happy to write some tests if you're feeling tied up. :-)

**Comment by jimbolla** [comment]
@vhmth Unfortunately I have commitments this weekend, so I won't be able to work on it again until Monday evening. I have off Tuesday and plan to work on it then as well.

I started writing a test but am having difficulty reproducing the behavior from scratch. I'm going to take a different approach and start with @Restuta's project and just start ripping out all the unrelated code until I'm left with the minimum code that reproduces the error. Then I'll turn that into a test.

**Comment by vhmth** [comment]
Sounds like a plan @jimbolla. If I have some time later on Sunday, I'll see if I can repro too. Maybe that will save ya some time and you could turn that into a test if I come up with that minimal example.

**Comment by jimbolla** [comment]
That sounds like it would be very helpful. Thanks. :smile:

**Comment by Restuta** [comment]
Don't hesitate to send a PR with minimal repro setup for the branch I created @vhmh

**Comment by vhmth** [comment]
Hey guys I'm sorry about the push off. A lot of stuff landed on my plate (life stuff) this weekend and couldn't get to working on repro-ing that bug. I will try tomorrow afternoon. I know @jimbolla said he wouldn't be able to get to it until tomorrow too, but it may be good to have two set of eyes on trying to repro it.

**Comment by jimbolla** [comment]
New PR #472 with fix and test is ready.

**Comment by timdorr** [comment]
Fixed in #472. I'll push out a beta release tonight.

**Comment by timdorr** [comment]
`react-redux-5.0.0-beta.1` has been published to `react-router@next`!

**Comment by Restuta** [comment]
on it, testing

**Comment by jimbolla** [comment]
Thanks, Tim!

**Comment by Restuta** [comment]
looking good so far, thanks guys!

**Comment by vhmth** [comment]
Been using it in our staging extension and things are looking good. Will be rolling out our extension using this with our 150-ish beta users and let y'all know if we run into any issues.

**Comment by SandHani45** [comment]
warning.js?6327:36 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the DrawingTool component.

hi please help out this issue im getting this error when i leave the component with out loading
